src/Tests/InstrEngineTests/InstrEngineTests/TargetAppCompiler.cs

```diff
@@ -68,50 +69,69 @@ internal class TargetAppCompiler
 
         internal static void CompileTestCode(string directoryPath)
         {
-            CompileCSharpTestCode(directoryPath);
+#if NETCOREAPP
```

Comment:
I think this needs to be even stricter... ref fields are not supported in any runtime version other than .NET 7. So this should be `NET7_0_OR_GREATER`.